### PR TITLE
Use supergraph schema to extract auth info

### PR DESCRIPTION
**Pull request by tninesling**
Use supergraph schema to extract auth info as auth information may not be available on the query planner's subgraph schemas. This undoes the auth changes made in #4975.

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [ ] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

File: apollo-router/src/query_planner/bridge_query_planner.rs [hunk 1]
```diff
@@ -1062,11 +1062,11 @@ impl QueryPlan {
 
     fn extract_authorization_metadata(
         &mut self,
-        subgraph_schemas: &SubgraphSchemas,
+        schema: &apollo_compiler::Schema,
```
Comment (Geal) [inline apollo-router/src/query_planner/bridge_query_planner.rs:1065]: isn't `supergraph_schema()` returnin a `Valid<Schema>`?

Comment (tninesling) [reply apollo-router/src/query_planner/bridge_query_planner.rs:1065]: It does. This first pass just restores the previous type signature, but I can update to use `Valid<Schema>` so we don't need `Valid::assume_valid_ref` anymore

Comment (tninesling) [reply apollo-router/src/query_planner/bridge_query_planner.rs]: Updated in https://github.com/apollographql/router/pull/5047/commits/cf25e1660cafe6eb323845f831a7eb546907c9e3

**Review by Geal — APPROVED** [review]

File: apollo-router/src/query_planner/fetch.rs [hunk 2]
```diff
@@ -627,13 +627,18 @@ impl FetchNode {
 
     pub(crate) fn extract_authorization_metadata(
         &mut self,
-        subgraph_schemas: &SubgraphSchemas,
+        schema: &Valid<apollo_compiler::Schema>,
         global_authorisation_cache_key: &CacheKeyMetadata,
     ) {
-        let doc = self.parsed_operation(subgraph_schemas);
-        let schema = &subgraph_schemas[self.service_name.as_str()];
+        let doc = ExecutableDocument::parse(
```
Comment (SimonSapin) [inline apollo-router/src/query_planner/fetch.rs:633]: Re parsing here is unfortunate. Would it make sense to call `self.parsed_operation()` here with the subgraph schema, but then pass the supergraph schema to `AuthorizationPlugin::generate_cache_metadata` for finding directives?

Comment (garypen) [reply apollo-router/src/query_planner/fetch.rs:633]: This was resolved by off-GH discussion. I'm going to merge.

**Pull request merged**